dds/DCPS/XTypes/DynamicData.cpp

```diff
@@ -0,0 +1,1068 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#include "DynamicData.h"
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace XTypes {
+
+DDS::ReturnCode_t DynamicData::get_descriptor(MemberDescriptor& value, MemberId id) const
+{
+}
+
+DDS::ReturnCode_t DynamicData::set_descriptor(MemberId id, const MemberDescriptor& value)
+{
+}
+
+bool DynamicData::equals(const DynamicData& other) const
+{
+}
+
+MemberId DynamicData::get_member_id_by_name(String name) const
+{
+}
+
+MemberId DynamicData::get_member_id_at_index(ACE_CDR::ULong) const
+{
+}
+
+ACE_CDR::ULong DynamicData::get_item_count() const
+{
+}
+
+DDS::ReturnCode_t DynamicData::clear_all_values()
+{
+}
+
+DDS::ReturnCode_t DynamicData::clear_nonkey_values()
+{
+}
+
+DDS::ReturnCode_t DynamicData::clear_value(MemberId id)
+{
+}
+
+DynamicData DynamicData::loan_value(MemberId id)
+{
+}
+
+DDS::ReturnCode_t DynamicData::return_loaned_value(const DynamicData& value)
+{
+}
+
+DynamicData DynamicData::clone() const
+{
+}
+
+DDS::ReturnCode_t DynamicData::get_int32_value(ACE_CDR::Long& value, MemberId id) const
+{
+  const OPENDDS_MAP<MemberId, size_t>::const_iterator it = offset_lookup_table_.find(id);
+  if (it != offset_lookup_table_.cend()) {
+    size_t offset = it->second;
+    std::ostringstream err_msg;
+    err_msg << "Failed to skip to the member with ID " << id;
+    if (!skip("get_int32_value", err_msg.str().c_str(), offset)) {
+      return DDS::RETCODE_ERROR;
+    }
+
+    if (!(strm_ >> value)) {
+      if (DCPS_debug_level >= 10) {
+        ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) DynamicData::get_int32_value -")
+                   ACE_TEXT(" Failed to read an int32 member with ID %d\n"), id));
+      }
+      return DDS::RETCODE_ERROR;
+    }
+
+    reset_rpos();
+    return DDS::RETCODE_OK;
+  }
+
+  DynamicTypeMember_rch member;
+  DDS::ReturnCode_t retcode = type_->get_member(member, id);
+  if (retcode != DDS::RETCODE_OK) {
+    if (DCPS_debug_level >= 10) {
+      ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) DynamicData::get_int32_value -")
+                 ACE_TEXT(" Failed to get DynamicTypeMember with ID %d\n"), id));
+    }
+    return retcode;
+  }
+
+  MemberDescriptor member_desc;
+  member->get_descriptor(member_desc);
+
+  if (member_desc.type->get_kind() != TK_INT32) {
+    if (DCPS_debug_level >= 10) {
+      ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) DynamicData::get_int32_value -")
+                 ACE_TEXT(" Member with ID %d is not an int32\n"), id));
```

Comment:
Maybe log which kind it is, would help logging. Maybe even log kind as a readable string